llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -117,9 +117,13 @@ bool llvm::isVectorIntrinsicWithScalarOpAtArg(Intrinsic::ID ID,
                                               unsigned ScalarOpdIdx) {
   switch (ID) {
   case Intrinsic::abs:
+  case Intrinsic::vp_abs:
   case Intrinsic::ctlz:
+  case Intrinsic::vp_ctlz:
   case Intrinsic::cttz:
+  case Intrinsic::vp_cttz:
   case Intrinsic::is_fpclass:
+  case Intrinsic::vp_is_fpclass:
   case Intrinsic::powi:
     return (ScalarOpdIdx == 1);
```

Comment:
In the example,
```
%r = call <4 x i32> @llvm.vp.abs.v4i32(<4 x i32> %a, <4 x i1> %mask, i32 %evl, i1 false)
```
 the scalar operands are `%evl` and `false` , located at indexes 2 and 3.
It looks like VectorUtil doesn't work well with vp intrinsics.
I’m unsure of the best practice here, but in #115274, I handle `%evl` by addng
```
 if (VPIntrinsic::isVPIntrinsic(ID) &&
      (ScalarOpdIdx == VPIntrinsic::getVectorLengthParamPos(ID)))
    return true;
```

Reply:
In fact, without building Vector-predicate Intrinsisc，the vp args sequential: 
![image](https://github.com/user-attachments/assets/55f08c55-2361-4bad-b4c7-45d428040d4d)

Reply:
Seem like the document need to update! 🫠 
```
commit 419948fe6708021524f86e15d755719d634ab8d2

Refs: llvmorg-17-init-2994-g419948fe6708
Author:     Yeting Kuo <yeting.kuo@sifive.com>
AuthorDate: Wed Feb 22 14:50:11 2023 +0800
Commit:     Yeting Kuo <yeting.kuo@sifive.com>
CommitDate: Thu Feb 23 13:58:21 2023 +0800

    [VP] Reorder is_int_min_poison/is_zero_poison operand before mask for vp.abs/ctlz/cttz.

    The patch ensures last two operands of vp.abs/ctlz/cttz are mask and evl.

    Reviewed By: craig.topper

    Differential Revision: https://reviews.llvm.org/D144536
```